### ISSUE TITLE
Image transform array doesn't use $configOverrides

### ISSUE THREAD

**Issue by fyrebase** [aelvan/Imager-Craft#6]
When passing a transform array to imager `$configOverrides` are not applied to the transform.
Take the following for example, only the width and height is applied to each transform.

```
{% set processedPhoto = craft.imager.transformImage(photo,
	[
		{ width: 1180, height: 680 },
		{ width: 900, height: 598 },
		{ width: 600, height: 399 },
		{ width: 400, height: 266 }
	],
	{
		mode: 'crop',
		mode: 'crop',
		position: '50% 50%'
	}
) %}
```

I've hacked in `$t = array_merge($t, $configOverrides);` at line 195 in ImagerService.php to make it work, but has the side effect of generating multiple images with a different hash.

**Comment by aelvan** [comment]
What you mean is that config overrides aren't taken into account when creating the filename? Or doesn't the mode and position work at all?

**Comment by fyrebase** [comment]
Mode and position aren't applied to the transform. I noticed it when I was applying a watermark to multiple images and it wasn't appearing on the transformed image.

**Comment by aelvan** [comment]
That's weird, I'm pretty sure I'm using that on several sites. I'll have a look at it.

**Comment by aelvan** [comment]
Sorry, I didn't really think when I answered. :) I think this actually works as intended. The thing is, the second parameter was suppose to be config overrides, ie only the settings that are in the config file is possible to override. Also, these settings doesn't modify the filename, since that would mean all config setting variables would need to be added to the filename. 

What I realize now, is that I should've had three parameters, also one for transform overrides, which was added to all the transforms in the first parameter. I'll have to think about how to add this, keep backwards compability and keep the syntax nice.
